### PR TITLE
Improved TaggableMatcherPrinterCommand outputs

### PR DESCRIPTION
**Pull request by lucaspcram**
### Description:
Does what it says on the tin. Removed extraneous newlines, redundant re-prints, and confusing formatting.

### Potential Impact:
Better output.

### Unit Test Approach:
Updated to reflect new output.

### Test Results:
👍

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

**Review by AlexHsieh22 — APPROVED** [review]
Nice cleanup!

**Review by matthieun — DISMISSED** [review]
Nice!

File: src/main/java/org/openstreetmap/atlas/tags/filters/matcher/README.md [hunk 1]
```diff
@@ -266,22 +264,32 @@ at the command line, and then pass as many `TaggableFilter` definitions as you w
...
+you want help debugging one. Make use of your shell's command substitution feature like so:
+```
+atlas print-matcher "$(atlas print-matcher --reverse 'foo->bar||baz->bat&cat->hat')"
+                                &
+                ┌───────────────┴───────────────┐
+                |                               =
+        ┌───────┴───────┐               ┌───────┴───────┐
+        =               =              cat             hat
+    ┌───┴───┐       ┌───┴───┐
+   foo     bar     baz     bat
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/tags/filters/matcher/README.md:292]: just gotta throw it out there that these printouts are gorgeous

File: src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/TaggableMatcherPrinterCommand.java [hunk 2]
```diff
@@ -108,7 +106,12 @@ private void executeDefaultContext()
                     throw exception;
                 }
             }
-            this.getCommandOutputDelegate().printlnStdout("");
+
+            // Print an extra newline between trees, but not for the last tree
+            if (i < definitions.size() - 1)
```
Comment (adahn6) [inline src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/TaggableMatcherPrinterCommand.java:111]: I was wondering why the switch in loop mechanics! 👍

**Review by adahn6 — APPROVED** [review]

**Pull request merged**